deepchecks/nlp/text_data.py

```diff
@@ -149,31 +153,59 @@ def copy(self: TDataset, rows_to_use: t.Optional[t.Sequence[int]] = None) -> TDa
             The rows to use in the new copy. If None, the new copy will contain all the rows.
         """
         cls = type(self)
-        logger_state = get_logger().disabled
-        get_logger().disabled = True  # Make sure we won't get the warning for setting class in the non multilabel case
-        if rows_to_use is None:
-            new_copy = cls(raw_text=self._text, tokenized_text=self._tokenized_text, label=self._label,
-                           task_type=self._task_type.value, name=self.name)
-            metadata, properties = self._metadata, self._properties
-            index_kept = self._original_text_index
-        else:
+
+        # TODO:
+        # why do we disable the root logger here instead of disabling/having
+        # a dedicated logger for this module/sub-package?
+        #
+        # NOTE:
+        # Make sure we won't get the warning for setting class in the non multilabel case
+        with disable_deepchecks_logger():
```

Comment:
@yromanyshyn as I understand we disabled the global logger since issues were raised from other dependent packages. So I'm not sure that disabling deepchecks logger would do that same.

Reply:
nevermind this note it is misleading, sorry

we disable the deepchecks logger, I did not change logic, it was this way, 
I just moved that code to the context manager